File: fixtures/differential-oracle-node/approval.js
// PLANTED-FLAWED command-approval control (raeuberkrebs canary fixture — NOT production code).
//
// This toy security-decision function auto-approves any `command <x>` invocation as "safe". That is
// a faithful miniature of a real carrier bypass: the `command` shell builtin runs its arguments as a
// command, so `command echo <marker>` is BELIEVED safe here yet actually EXECUTES. The
// differential-oracle lane must catch that divergence (belief=safe, marker fires) to prove itself live.
export function isCommandSafe(cmd) {
  if (typeof cmd !== "string") return false;
  // FLAW: trusts the `command` builtin without unwrapping the inner command it runs.
  if (cmd.startsWith("command ")) return true;
  // A correctly-modeled safe case: a bare `true` runs nothing dangerous.
  return cmd === "true";
}

File: fixtures/differential-oracle-dotnet/Approval.cs
using System;

namespace RkFixture
{
    // PLANTED VULNERABILITY (do not "fix"): a command-approval control that auto-approves any
    // `command <x>` invocation — a miniature of the real openclaw `command`-builtin carrier bypass
    // (GHSA-ghpx). The control BELIEVES the `command ` prefix is a safe no-op, but `command echo X`
    // actually RUNS echo. This fixture exists so the C# command-approval differential-oracle lane's
    // canary can prove itself LIVE: the lane MUST report a divergence here (believed-safe AND the
    // marker executed), or the lane is quarantined (fail-closed). `echo` keeps it benign.
    public static class Approval
    {
        public static bool IsCommandSafe(string cmd)
        {
            if (cmd == null) return false;
            // FLAW: the `command` builtin prefix is whitelisted as harmless, but it is a CARRIER —
            // `command echo <marker>` runs echo. The oracle drives exactly this divergence.
            if (cmd.StartsWith("command ")) return true;
            // Everything else is only "safe" if it is a literal no-op — so the plain `echo <marker>`
            // corpus item is (correctly) judged UNSAFE and must NOT be reported (no false positive).
            return cmd == "true" || cmd == ":";
        }
    }
}
